sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ConnectionPolicy.java

```diff
@@ -475,6 +478,15 @@ public ConnectionPolicy setPreferredRegions(List<String> preferredRegions) {
         return this;
     }
 
+    public ConnectionPolicy setExcludedRegionsSupplier(Supplier<CosmosExcludedRegions> excludedRegionsSupplier) {
```

Comment:
I think you need to use an AtomicReference here as well - otherwise an update to the Supplier would not be reflected in the connectionPolicy?

Reply:
nvm - got confused